main/views.py

```diff
@@ -1,3 +1,127 @@
+
+    def form_valid(self, form):
+        user = form.save()
+        login(self.request, user)
+        return redirect('index')
+
+
+class IndexView(ListView):
+    template_name = 'index.html'
+    context_object_name = 'sites'
+
+    def get_context_data(self, **kwargs):
+        context = super(IndexView, self).get_context_data(**kwargs)
+        context['form'] = self.request.GET.get('form', SiteToCheckForm)
+        return context
+
+    def get_queryset(self):
+        return SiteToCheck.objects.filter(user=self.request.user)
+
+
+class AddSiteToCheckView(FormView):
+    template_name = 'index.html'
+    context_object_name = 'sites'
+    model = SiteToCheck
+    form_class = SiteToCheckForm
+
+    def form_valid(self, form):
+        cd = form.cleaned_data
+        if SiteToCheck.objects.filter(url=cd['url'], user=self.request.user).exists():
+            messages.error(self.request, 'The page already exists in database')
+        else:
+            data = SiteDownChecker(cd['url'], user=self.request.user).status()
+            success_message_text = f"The page hes been added. \n\nStatus: {data['last_status']}, Response time: \
+            {data['last_response_time']}"
+            messages.success(self.request, success_message_text)
+        return redirect('/')
+
+    def form_invalid(self, form):
+        messages.error(self.request, 'Please enter the correct URL')
+        return redirect('/')
+
+
+class SiteDetailView(DetailView):
+    model = SiteToCheck
+    template_name = 'details.html'
+    context_object_name = 'url'
+
+
+class SiteDeleteView(DeleteView):
+    model = SiteToCheck
+    success_url = reverse_lazy('index')
+
+    def get(self, *args, **kwargs):
+        message = f'{self.get_object().url} was deleted'
+        messages.success(self.request, message)
+        return self.post(*args, **kwargs)
+
+
+@login_required
+def update_email(request):
+    modify_email(request)
+    return redirect('/')
+
+
+@login_required
+def url_refresh(request, pk):
+    url = get_object_or_404(SiteToCheck, pk=pk, user=request.user)
+    if request.method == 'GET':
+        data = SiteDownChecker(url.url, user=request.user).status()
+        success_message_text = f"{url} Status: {data['last_status']}, Response time: {data['last_response_time']}"
+        messages.success(request, success_message_text)
+        return redirect(f'/details/{pk}')
+    return render(request, 'index.html')
+
+
+@login_required
+def check_all(request):
+    sites = SiteToCheck.objects.filter(user=request.user)
+    for url in sites:
+        SiteDownChecker(url, request.user).status()
+    messages.success(request, 'Successfully refreshed sites statuses')
+    return redirect('/')
+
+
+def modify_settings(request):
+    if request.method == 'POST':
+        response_json = request.POST
+        response_json = json.dumps(response_json)
+        data = json.loads(response_json)
+        if data['id'] == 'proxy':
+            if config.PROXY is True:
```

Comment:
Czyli upraszczając config.PROXY = not config.PROXY. Toglujesz tę zmienną na przeciwny stan :)

Reply:
Tutaj rozumiem, że powinno być po prostu:
`if config.PROXY:`